test/scenarios.py

```diff
@@ -7227,6 +7227,174 @@
+        # for both peers. However we don't have such posibitlity in our test
+        # framework to test things across containers
+        {
+            "DROP_PEER_REN": ActionList(
+                [
+                    action.WaitForAnotherPeer("DROP_PEER_STIMPY"),
+                    action.ConfigureNetwork(rate="200mbit"),
+                    action.Start("DROP_PEER_REN", dbpath="/tmp/db/29-14-ren.sqlite"),
+                    action.NewTransfer(
+                        "DROP_PEER_STIMPY",
+                        [
+                            "/tmp/testfile-bulk-01",
+                            "/tmp/testfile-bulk-02",
+                            "/tmp/testfile-bulk-03",
+                            "/tmp/testfile-bulk-04",
+                        ],
+                    ),
+                    action.Wait(
+                        event.Queued(
+                            0,
+                            "DROP_PEER_STIMPY",
+                            {
+                                event.File(
+                                    FILES["testfile-bulk-01"].id,
+                                    "testfile-bulk-01",
+                                    10485760,
+                                ),
+                                event.File(
+                                    FILES["testfile-bulk-02"].id,
+                                    "testfile-bulk-02",
+                                    10485760,
+                                ),
+                                event.File(
+                                    FILES["testfile-bulk-03"].id,
+                                    "testfile-bulk-03",
+                                    10485760,
+                                ),
+                                event.File(
+                                    FILES["testfile-bulk-04"].id,
+                                    "testfile-bulk-04",
+                                    10485760,
+                                ),
+                            },
+                        )
+                    ),
+                    action.WaitRacy(
+                        [
+                            event.Start(0, FILES["testfile-bulk-01"].id),
+                            event.Start(0, FILES["testfile-bulk-02"].id),
+                            event.Start(0, FILES["testfile-bulk-03"].id),
+                            event.Start(0, FILES["testfile-bulk-04"].id),
+                        ]
+                    ),
+                    action.WaitForOneOf(
+                        [
+                            event.FinishFileUploaded(0, FILES["testfile-bulk-01"].id),
+                            event.FinishFileUploaded(0, FILES["testfile-bulk-02"].id),
+                            event.FinishFileUploaded(0, FILES["testfile-bulk-03"].id),
+                            event.FinishFileUploaded(0, FILES["testfile-bulk-04"].id),
+                        ]
+                    ),
```

Comment:
Wouldn't it be a bit safer for reproduction to wait for all of the files to be Uploaded and the nstop? We're not racing to do the stopping as early as possible. this way we should have a higher chance of catching the issue.

Not that anything is bad with the current setup though

Reply:
That's not true. We need some of them to be `not finished` on the sender's side but finished on the receiver's. If all are finished on the sender's side then there's nothing to mismatch. That's the point

Reply:
Wasn't initially clear that was the case